src/client/mod.rs

```diff
@@ -137,9 +138,35 @@ impl Client {
 
         full_client.get_history();
 
+        //Start listening for Events
+        full_client.listen().await;
+
         Ok(full_client)
     }
 
+    /// Listen for incoming events(messages) via IncomingConnections.
+    pub async fn listen(&mut self) {
```

Comment:
I dont think this needs to be a pub api, no? Can just be used by client only

Reply:
Thought we might need it pub when we have Apps, but yea, priv would make the most sense now :+1: